cvat/apps/dataset_manager/views.py

```diff
@@ -127,14 +176,55 @@ def export_project_annotations(project_id, dst_format=None, server_url=None):
     return export(dst_format, project_id=project_id, server_url=server_url, save_images=False)
 
 
-def clear_export_cache(file_path, file_ctime, logger):
+def clear_export_cache(file_path: str, file_ctime: float, logger: logging.Logger) -> None:
     try:
-        if osp.exists(file_path) and osp.getctime(file_path) == file_ctime:
+        with get_dataset_cache_lock(
+            file_path,
+            block=True,
+            acquire_timeout=EXPORT_CACHE_LOCK_TIMEOUT,
+            ttl=rq.get_current_job().timeout,
+        ):
+            if 'job' in file_path:
+                instance_type = 'job'
+            elif 'task' in file_path:
+                instance_type = 'task'
+            elif 'project' in file_path:
+                instance_type = 'project'
+            else:
+                assert False
+
+            cache_ttl = get_export_cache_ttl(instance_type)
+
+            instance_timestamp = None
+            with suppress(AttributeError): # backward compatibility
```

Comment:
Can you elaborate on the "backwards compatibility" comment? How could an `AttributeError` occur here?

Reply:
It can come from `match.group()`, if `match` is `None`, i.e. if the file name doesn't contain the added fragment with timestamp. This can happen for jobs created before the patch. I'm not 100% sure it is a problem with inmemory redis.

Reply:
> I'm not 100% sure backward compatibility is a problem with inmemory redis.

FWIW, we do have persistence enabled by default for in-memory Redis, so maintaining backwards compatibility is useful.

Reply:
Ok, good to know. In this case, these efforts make sense. Maybe I will add a test for this.